lib/model/migrations/20240322-01-add-entity-source-index-to-audits.js

```diff
@@ -0,0 +1,24 @@
+// Copyright 2024 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const up = async (db) => {
+  await db.raw('DROP INDEX audits_details_entity_index');
+  await db.raw("CREATE INDEX audits_details_entity_def_index ON audits USING HASH (((details ->> 'entityDefId')::INTEGER))");
+  await db.raw("CREATE INDEX audits_details_entity_source_index ON audits USING HASH (((details ->> 'sourceId')::INTEGER))");
```

Comment:
It looks like the query uses `->`, not `->>`:

https://github.com/getodk/central-backend/blob/428ebdd06143684759f1507a708db69fd2f0b76c/lib/model/query/audits.js#L139

It might be a good idea for the index to exactly match the query. We could either change the index to match the query or change the query to match the index.